packages/langium/src/lsp/rename-refactoring.ts

```diff
@@ -0,0 +1,68 @@
+/******************************************************************************
+ * Copyright 2021 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import { Range, RenameParams, TextDocumentPositionParams, TextEdit, WorkspaceEdit } from 'vscode-languageserver';
+import { Position } from 'vscode-languageserver-textdocument';
+import { LangiumDocument } from '../documents/document';
+import { isNamed, NameProvider } from '../references/naming';
+import { References } from '../references/references';
+import { LangiumServices } from '../services';
+import { CstNode } from '../syntax-tree';
+import { findLeafNodeAtOffset } from '../utils/ast-util';
+import { toRange } from '../utils/cst-util';
+import { ReferenceFinder } from './reference-finder';
+
+export interface RenameHandler {
+    renameElement(document: LangiumDocument, params: RenameParams): WorkspaceEdit | undefined;
+    prepareRename(document: LangiumDocument, params: TextDocumentPositionParams): Range | undefined;
+}
+
+export class DefaultRenameHandler implements RenameHandler {
+
+    protected readonly referenceFinder: ReferenceFinder
+    protected readonly references: References
+    protected readonly nameProvider: NameProvider
+
+    constructor(services: LangiumServices) {
+        this.referenceFinder = services.lsp.ReferenceFinder;
+        this.references = services.references.References;
+        this.nameProvider = services.references.NameProvider;
+    }
+    renameElement(document: LangiumDocument, params: RenameParams): WorkspaceEdit | undefined {
+        const textChanges: { [uri: string]: TextEdit[] } = {};
+        this.referenceFinder.findReferences(document, params, true).forEach((location) => {
+            textChanges[location.uri] ?
+                textChanges[location.uri].push(TextEdit.replace(location.range, params.newName))
+                :
+                textChanges[location.uri] = [TextEdit.replace(location.range, params.newName)];
+        });
+        return { changes: textChanges };
+    }
```

Comment:
Nitpick: The formatting here looks a bit off
```suggestion
    }
    
    renameElement(document: LangiumDocument, params: RenameParams): WorkspaceEdit | undefined {
        const changes: Record<string, TextEdit[]> = {};
        this.referenceFinder.findReferences(document, params, true).forEach(location => {
            textChanges[location.uri]
                ? textChanges[location.uri].push(TextEdit.replace(location.range, params.newName))
                : textChanges[location.uri] = [TextEdit.replace(location.range, params.newName)];
        });
        return { changes };
    }
```

Reply:
Here is more than just formatting changed :-/